src/Testcontainers.Gitlab/GitlabContainer.cs

```diff
@@ -0,0 +1,93 @@
+{
+    /// <summary>
+    /// The password for the root user
+    /// </summary>
+    public string RootPassword { get; private set; }
+
+    public override async Task StartAsync(CancellationToken ct = default)
+    {
+        await base.StartAsync(ct);
+        RootPassword = await GetInitialRootPassword();
+    }
+
+    /// <summary>
+    /// Generate a personal access token.
+    /// </summary>
+    /// <param name="pat">The personal access token to create.</param>
+    /// <returns></returns>
+    /// <exception cref="DataMisalignedException"></exception>
+    public async Task<PersonalAccessToken> GenerateAccessToken(PersonalAccessToken pat)
+    {
+        var scope = "[" + '\'' + pat.Scope.ToString().Replace(", ", "\', \'") + '\'' + "]";
+
+        var command = $"token = User.find_by_username('{pat.User}')" +
+            $".personal_access_tokens" +
+            $".create(name: '{pat.Name}', scopes: {scope}, expires_at: {pat.ExpirationInDays}.days.from_now); " +
+            $"puts token.cleartext_tokens";
+
+        var tokenCommand = new List<string>{
+            { "gitlab-rails" },
+            { "runner" },
+            { command }
+        };
+
+        ExecResult tokenResult = await ExecAsync(tokenCommand);
+
+        string token = "";
+        if (tokenResult.ExitCode == 0)
+        {
+            var match = GitlabRegex.GitlabPersonalAccessToken.Match(tokenResult.Stdout);
+            token = match.Value;
+        }
+        else
+        {
+            throw new DataMisalignedException("Stderr: " + tokenResult.Stderr + "|" + "Stdout: " + tokenResult.Stdout);
+        }
+        pat.TokenInternal = token;
+        return pat;
+    }
+
+    /// <summary>
+    /// Generate a personal access token.
+    /// </summary>
+    /// <param name="name">Name of the personal access token. If left empty a GUID will be used.</param>
+    /// <param name="user">The name of the user that owns this personal access token.</param>
+    /// <param name="scope">The scope that will be given to the token.</param>
+    /// <param name="expirationInDays">Days until the tokens expires.</param>
+    /// <returns></returns>
+    public async Task<PersonalAccessToken> GenerateAccessToken(string user, PersonalAccessTokenScopes scope, string name = "", int expirationInDays = 365)
+        => await GenerateAccessToken(new PersonalAccessToken
+        {
+            Name = string.IsNullOrWhiteSpace(name) ? Guid.NewGuid().ToString() : name,
+            User = user,
+            Scope = scope,
+            ExpirationInDays = expirationInDays
+        });
+
+
+    /// <summary>
+    /// This method returns the initial root password for the gitlab root user.
+    /// </summary>
+    /// <returns>Returns the initial root password generated by gitlab.</returns>
+    private async Task<string> GetInitialRootPassword()
+    {
+        var byteArray = await ReadFileAsync("/etc/gitlab/initial_root_password");
```

Comment:
Testcontainers allows uploading files before the container starts. Can we upload the `initial_root_password` file to specify the password? Can we simply use basic authentication? This would make the module simpler and should be enough for testing purposes (I guess).

Reply:
Yeah I checked last night if there is some easier method to provide a root password and there a sereval ways. You can set the root password with an environment variable named `GITLAB_ROOT_PASSWORD` ([see here for dokumentation](https://docs.gitlab.com/ee/administration/environment_variables.html)). Another way would be to provide a file named `root_password.txt` in the root directory. This is documented [here](https://docs.gitlab.com/ee/install/docker.html#install-gitlab-using-docker-swarm-mode).

I would strongly prefer to set the root password via an environment variable, would either way would be valid. I try to find some time this week to resolve this. 

If you have an preferred implementation than just let me know.

Reply:
Should be fixed now. I've selected the approach via an environment variable just to keep it simple.